pkg/cache/cache.go

```diff
@@ -188,6 +188,12 @@ type Options struct {
 	// unless there is already one set in ByObject or DefaultNamespaces.
 	DefaultTransform toolscache.TransformFunc
 
+	// DefaultWatchErrorHandler will be used to the WatchErrorHandler which is called
+	// whenever ListAndWatch drops the connection with an error.
+	//
+	// After calling this handler, the informer will backoff and retry.
+	DefaultWatchErrorHandler toolscache.WatchErrorHandler
```

Comment:
Two questions:
1. What does this default to inside the `SharedIndexInformer`, a logger?
2. The infomers relist once in a while. Does this get called on that part? The reason I am asking is that I am wondering if it would be ok to have an implementation here that exits the binary with an error or if that would break things.

Reply:
> What does this default to inside the SharedIndexInformer, a logger?

From my understanding it seems to just log even in the case where `utilruntime.HandleError` is the default case e.g io.EOF, etc

> The infomers relist once in a while. Does this get called on that part? The reason I am asking is that I am wondering if it would be ok to have an implementation here that exits the binary with an error or if that would break things.

From what I understand, the relist calls this method during the relist but because it exited normally, it wouldn't error but log to notify.  

It seems like we were already using this default behavior but not exposing a way to `SetErrorWatchHandler` for people to set on the SharedIndexInformer.  Initially I was setting this to `DefaultWatchErrorHandler` but because the behavior of the SharedIndexInformer is to use this if not set, feedback was presented to not set this as it was redundant.